### ISSUE TITLE
Collating captions produces zeros

### ISSUE THREAD

**Issue by ramanishka** [XiaoxiaoGuo/fashion-iq#2]
These lines apparently copy zeroes from the newly created tensor instead of actual token ids:

https://github.com/XiaoxiaoGuo/fashion-iq/blob/529a8b9a8122ed856a3252e37184316e424bc954/start_kit/data_loader.py#L89-L92

**Comment by YupengGao** [comment]
I don't think there is a problem.

**Comment by ramanishka** [comment]
@YupengGao Correct me if I'm wrong but it takes original captions and returns newly created tensor with zeros. This data loader ignores captions.

```
    target_images, candidate_images, captions, meta = zip(*data)
    ...
    captions = torch.zeros(len(captions), max(lengths)).long()
    for i, cap in enumerate(captions):
        end = lengths[i]
        captions[i, :end] = cap[:end]
    ...
    return target_images, candidate_images, captions, lengths, meta
```

**Comment by YupengGao** [comment]
@ramanishka Thanks for spotting the bug. I have fixed it.

**Comment by zxh19976666** [comment]
How to divide the datasets into training set and test set，I have download all images，but how to divide it？

**Comment by zxh19976666** [comment]
> These lines apparently copy zeroes from the newly created tensor instead of actual token ids:
> 
> https://github.com/XiaoxiaoGuo/fashion-iq/blob/529a8b9a8122ed856a3252e37184316e424bc954/start_kit/data_loader.py#L89-L92

How to divide the datasets into training set and test set，I have download all images，but how to divide it？
